jdisc-cloud-aws/src/main/java/ai/vespa/secret/aws/AsmSecretStoreBase.java

```diff
@@ -86,7 +93,6 @@ public void deconstruct() {
 
     // Only for testing
     public Set<AwsRolePath> clientRoleNames() {
```

Comment:
Suggest to return a set of AssumedRoleInfo (and rename, I see I forgot) and add verification in AsmTenantSecretReaderTest:74